tensorflow/lite/micro/tools/make/targets/cortex_m_corstone_300_makefile.inc

```diff
@@ -0,0 +1,180 @@
+endif
+
+FLOAT := soft
+GCC_TARGET_ARCH := $(TARGET_ARCH)
+
+ifeq ($(TARGET_ARCH), cortex-m0)
+  CORE=M0
+
+else ifeq ($(TARGET_ARCH), cortex-m3)
+  CORE=M3
+
+else ifeq ($(TARGET_ARCH), cortex-m33)
+  CORE=M33
+  FLOAT=hard
+  CMSIS_ARM_FEATURES := _DSP_DP
+
+else ifeq ($(TARGET_ARCH), cortex-m33+nodsp)
+  CORE=M33
+
+else ifeq ($(TARGET_ARCH), cortex-m4)
+  CORE=M4
+  GCC_TARGET_ARCH := cortex-m4+nofp
+
+else ifeq ($(TARGET_ARCH), cortex-m4+fp)
+  CORE=M4
+  FLOAT=hard
+  GCC_TARGET_ARCH := cortex-m4
+  CMSIS_ARM_FEATURES := _FP
+
+else ifeq ($(TARGET_ARCH), cortex-m55)
+  CORE=M55
+  FLOAT=hard
+
+else ifeq ($(TARGET_ARCH), cortex-m55+nodsp+nofp)
+  CORE=M55
+
+else ifeq ($(TARGET_ARCH), cortex-m55+nofp)
+  CORE=M55
+
+else ifeq ($(TARGET_ARCH), cortex-m7)
+  CORE=M7
+  GCC_TARGET_ARCH := cortex-m7+nofp
+
+else ifeq ($(TARGET_ARCH), cortex-m7+fp)
+  CORE=M7
+  FLOAT=hard
+  GCC_TARGET_ARCH := cortex-m7
+  CMSIS_ARM_FEATURES := _DP
+
+else
+  $(error "TARGET_ARCH=$(TARGET_ARCH) is not supported")
+endif
+
+ifneq ($(filter cortex-m55%,$(TARGET_ARCH)),)
+  # soft-abi=soft disables MVE - use softfp instead for M55.
+  ifeq ($(FLOAT),soft)
+    FLOAT=softfp
+  endif
+endif
+
+ifeq ($(TOOLCHAIN), gcc)
+  export PATH := $(MAKEFILE_DIR)/downloads/gcc_embedded/bin/:$(PATH)
+  DOWNLOAD_RESULT := $(shell $(MAKEFILE_DIR)/arm_gcc_download.sh ${MAKEFILE_DIR}/downloads)
+  ifneq ($(DOWNLOAD_RESULT), SUCCESS)
+    $(error Something went wrong with the GCC download: $(DOWNLOAD_RESULT))
+  endif
+  TARGET_TOOLCHAIN_PREFIX := arm-none-eabi-
+
+  FLAGS_GCC = -mcpu=$(GCC_TARGET_ARCH) -mfpu=auto
+  CXXFLAGS += $(FLAGS_GCC)
+  CCFLAGS += $(FLAGS_GCC)
+
+  # Run C preprocessor on linker file to get rid of ifdefs.
+  ifeq (,$(wildcard $(ETHOS_U_CORE_PLATFORM)/platform_parsed.ld))
+    RESULT := $(shell arm-none-eabi-gcc -E -x c -P -o $(ETHOS_U_CORE_PLATFORM)/platform_parsed.ld \
+      $(ETHOS_U_CORE_PLATFORM)/platform.ld)
+    ifneq ($(RESULT),)
+      $(error Something went wrong when parsing linker file: $(RESULT))
+    endif
+  endif
+
+  LDFLAGS += \
+    --specs=nosys.specs \
+    -T $(ETHOS_U_CORE_PLATFORM)/platform_parsed.ld \
+    -Wl,-Map=${TENSORFLOW_ROOT}$(MAKEFILE_DIR)/gen/$(TARGET).map,--cref \
+    -Wl,--gc-sections \
+    --entry Reset_Handler
+
+else
+  $(error "TOOLCHAIN=$(TOOLCHAIN) is not supported.")
+endif
+
+# TODO: fix warnings.
+OMIT_ERRORS = \
+  -Wno-implicit-fallthrough \
+  -Wno-strict-aliasing
+
+PLATFORM_FLAGS = \
+  -DTF_LITE_MCU_DEBUG_LOG \
+  -mthumb \
+  -mfloat-abi=$(FLOAT) \
+  -funsigned-char \
+  -mlittle-endian \
+  ${OMIT_ERRORS} \
+  -fomit-frame-pointer \
+  -MD \
+  -DCPU_$(CORE)=1
+
+# Common + C/C++ flags
+CXXFLAGS += $(PLATFORM_FLAGS)
+CCFLAGS += $(PLATFORM_FLAGS)
+
+ARM_CPU := $(subst cortex-m,ARMCM,$(GCC_TARGET_ARCH))
+ARM_CPU := $(subst +nofp,,$(ARM_CPU))
+CXXFLAGS += -D$(ARM_CPU)$(CMSIS_ARM_FEATURES)
+CCFLAGS += -D$(ARM_CPU)$(CMSIS_ARM_FEATURES)
+
+THIRD_PARTY_CC_SRCS += \
+  $(wildcard $(ETHOS_U_CORE_PLATFORM)/*.c)
```

Comment:
Perhaps it better to be explicit what files we include from core platform?

Reply:
Yes

Reply:
I don't see any changes for this comment. Just flagging in case you missed some changes when adding commits.

Reply:
I see the changes at lines 135-137

Reply:
I'm a bit confused. I see

```
THIRD_PARTY_CC_SRCS += \
￼  $(wildcard $(ETHOS_U_CORE_PLATFORM)/*.c)
```

and
```
THIRD_PARTY_CC_SRCS += \
￼  $(ETHOS_U_CORE_PLATFORM)/retarget.c \
￼  $(ETHOS_U_CORE_PLATFORM)/uart.c`
```

is this intentional?